### PR TITLE
workflow file for cherry-pick on comment

### LINKED ISSUE #3656: Add a provision in KServe repo to allow cherry pick of PRs to release branches
/kind feature

**Describe the solution you'd like**

Currently, when a new release branch is created, all the security patches and / or critical patches that were added to master have to manually merged to release branch. This is a painful process. It would be nice to have a mechanism, that could allow seamless merges in a semi-automated mechanism.

### PR DESCRIPTION
**Pull request by andyi2it**
**What this PR does / why we need it**:

This PR would allow cherry picking a PR and create a new PR on a target branch. Example, if a new critical fix was added to master, one can add a comment to cherry-pick to release-branch.
At this moment it is configured to allow only repo owner to cherry-pick.

### Usage:
`/cherry-pick release-branch`
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3656 

**Release note**:

```release-note
NONE
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

**Comment by israel-hdez** [comment]
There is already a cherry-pick workflow: https://github.com/kserve/kserve/blob/master/.github/workflows/cherry-pick.yml
I see it as outdated. Maybe, we should replace/update that one instead of creating a new file?

**Review by terrytangyuan — COMMENTED** [review]
This would be great to have.

**Comment by yuzisun** [comment]
@andyi2it Can you help add the instruction for how to cherry pick the commits to the release doc ?

**Comment by yuzisun** [comment]
> There is already a cherry-pick workflow: https://github.com/kserve/kserve/blob/master/.github/workflows/cherry-pick.yml I see it as outdated. Maybe, we should replace/update that one instead of creating a new file?

This one does work not well, we can delete it

File: .github/workflows/comment-cherry-pick.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  cherry-pick:
+    
+    name: Cherry Pick
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/cherry-pick') && github.event.comment.author_association == 'OWNER'
```
Comment (yuzisun) [inline .github/workflows/comment-cherry-pick.yml]: Let's change to kserve-contributors

**Comment by spolti** [comment]
> @andyi2it Can you help add the instruction for how to cherry pick the commits to the release doc ?

Maybe a small note in the `[CONTRIBUTING.md](https://github.com/kserve/kserve/blob/master/CONTRIBUTING.md)` as well

**Pull request merged**